packages/teuchos/core/src/Teuchos_ArrayView.hpp

```diff
@@ -315,13 +315,26 @@ T* ArrayView<T>::getRawPtr() const
   return ptr_;
 }
 
+template<class T> inline
+T* ArrayView<T>::data() const
+{
+  debug_assert_valid_ptr();
```

Comment:
To avoid duplication, why not just inline call `this->getRawPtr()`?

Reply:
@bartlettroscoe I wanted to keep the function call stack short so the compiler has a chance at inlining :-)

Reply:
@mhoemmen said:

> I wanted to keep the function call stack short so the compiler has a chance at inlining :-)

Okay, I can accept that.  Performance and  maintainability are often at odds. This is pretty minor duplication.  But it is fragile to someone removing the call to `debug_assert_valid_ptr()` and there are no unit tests that ensure that `data()` throws in debug mode for null or invalid pointers (like dangling references).  So technically, there is now a lack of testing.

Reply:
@bartlettroscoe Not true -- line 353 of `teuchos/core/test/MemoryManagement/ArrayView_UnitTests.cpp` tests that `.data()` throws in debug mode on dangling reference.

Reply:
Thanks!  I missed that looking over the tests.